prefect_docker/worker.py

```diff
@@ -372,6 +373,12 @@ class DockerWorker(BaseWorker):
     type = "docker"
     job_configuration = DockerWorkerJobConfiguration
 
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self._client = get_client()
+        if self._client.server_type == ServerType.EPHEMERAL:
+            raise RuntimeError("Docker worker cannot be used with an ephemeral server.")
```

Comment:
This should probably be done as a part of `setup` to avoid duplicate client creations.
```suggestion
    def setup(self):
        super().setup()
        if self._client.server_type == ServerType.EPHEMERAL:
            raise RuntimeError("Docker worker cannot be used with an ephemeral server.")
```